packages/offix-datastore/src/DataStore.ts

```diff
@@ -1,74 +1,77 @@
 import { Model, ModelConfig } from "./Model";
-import { LocalStorage } from "./storage";
-import { createGraphQLClient, GraphQLClientConfig } from "./replication/client/GraphQLClient";
-import { buildGraphQLCRUDQueries } from "./replication/graphqlcrud/buildGraphQLCRUDQueries";
+import { LocalStorage, StorageAdapter } from "./storage";
+import { createGraphQLClient } from "./replication/GraphQLClient";
+import { buildGraphQLCRUDQueries } from "./replication/documents/buildGraphQLCRUDQueries";
 import { IReplicator } from "./replication/api/Replicator";
-import { MutationReplicationEngine } from "./replication";
-import { GraphQLCRUDReplicator } from "./replication/graphqlcrud/GraphQLCRUDReplicator";
+import { GraphQLCRUDReplicator } from "./replication/GraphQLCRUDReplicator";
 import { IndexedDBStorageAdapter } from "./storage/adapters/IndexedDBStorageAdapter";
+import { DataStoreConfig } from "./DataStoreConfig";
+import { createLogger, enableLogger } from "./utils/logger";
+
+const logger = createLogger("DataStore");
+// TODO disable logging before release
+enableLogger();
+
+const metadataStoreName = "model_metadata"
 
 /**
- * Configuration Options for DataStore
+ * Custom implementation
  */
-export interface DataStoreConfig {
-  /**
-   * The Database name
-   */
-  dbName: string;
-
+export interface CustomEngines {
   /**
-   * The URQL config
+   * Custom storage adapter.
+   * By default DataStore will use IndexedDB that might not be available in every environment.
+   * If you wish to override adapter you can supply it here
    */
-  clientConfig: GraphQLClientConfig;
+  storeAdapter?: StorageAdapter,
 
   /**
-   * The Schema Version number. Used to trigger a Schema upgrade
+   * Custom replication mechanism that will replicate data.
+   * By default DataStore will be GraphQL (https://graphqlcrud.org) replication mechanism
    */
-  schemaVersion?: number;
+  replicator?: IReplicator
 }
 
+/**
+ * ____ Offix DataStore ___
+ * Client side storage with efficient querying capabilities
+ * and GraphQL replication engine.
+ */
 export class DataStore {
-  public networkStatus: any;
-  private dbName: string;
-  private schemaVersion: number;
   private models: Model<unknown>[];
-  private indexedDB: IndexedDBStorageAdapter;
   private storage: LocalStorage;
-  private clientConfig: any;
-  private metadataName = "metadata";
 
-  constructor(config: DataStoreConfig) {
-    this.dbName = config.dbName;
-    this.schemaVersion = config.schemaVersion || 1; // return 1 if schemaVersion is undefined or 0
-    this.clientConfig = config.clientConfig;
+  private config: DataStoreConfig;
```

Comment:
Key for datastore was to have clear architecture.

Datastore should have only:
- Models
- Store
- Replication

We can connect them later.